src/string_util.cpp

```diff
@@ -146,6 +156,11 @@ namespace libtorrent
 			*begin++ = printable[random(sizeof(printable) - 2)];
 	}
 
+	bool string_ends_with(std::string const& s1, std::string const& s2)
```

Comment:
what do you think about using string_ref here instead? that way the ".i2p" string literal would have to be heap allocated into a std::string